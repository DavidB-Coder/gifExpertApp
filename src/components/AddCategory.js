import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim()) {
            setCategory( ps => [inputValue,...ps] );        
            setInputValue('');
        }
    }


    return (
        <>
            <form onSubmit={ handleSubmit }>
                <input
                    type="text"
                    value={ inputValue }                
                    onChange={(e) => {setInputValue(e.target.value)}}
                    placeholder={'Ingresa una nueva categorÃ­a'}
                />
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategory : PropTypes.func.isRequired
};
