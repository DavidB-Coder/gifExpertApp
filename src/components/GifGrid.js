import React from 'react'
import { useFetchGifs } from '../Hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs( category );
    
    return (
        <>
            { loading && <p className="animate__animated animate__flash">Loading...</p> }
            <h3>{category}</h3>                        
            <div className="card-grid">
                {data.map((img) => {
                    return(
                        <GifGridItem 
                            key={img.id}                         
                            {...img}
                        />
                    )
                })}
            </div>            
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}