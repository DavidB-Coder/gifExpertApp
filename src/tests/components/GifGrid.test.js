import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../Hooks/useFetchGifs'
jest.mock('../../Hooks/useFetchGifs')

describe('Pruebas al componente GifGrid', () => {
    
    const category = "anyone"
    
    test('debe mostrar el contenido del componente', () => {
        useFetchGifs.mockReturnValue({
            data:[],
            loading:true
        });
        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [{
            id: 'abc',
            url: 'https://url.com.co',
            title: 'titulo'
        }]

        useFetchGifs.mockReturnValue({
            data:gifs,
            loading:false
        });

        const wrapper = shallow(<GifGrid category={category}/>)
        expect(wrapper.find('p').exists()).toBe(false)
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    });
});
