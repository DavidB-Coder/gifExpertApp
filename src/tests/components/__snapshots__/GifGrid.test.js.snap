// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGrid debe mostrar el contenido del componente 1`] = `
<Fragment>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <h3>
    anyone
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
